src/keymap-manager.coffee

```diff
@@ -409,15 +410,84 @@ class KeymapManager
   # target is `.defaultTarget` if that property is assigned on the keymap.
   #
   # * `event` A `KeyboardEvent` of type 'keydown'
-  handleKeyboardEvent: (event) ->
+  handleKeyboardEvent: (event, {replay, disabledBindings}={}) ->
+    # Handling keyboard events is complicated and very nuanced. The
+    # complexity is all because of multi-stroke bindings. For example:
+    #
+    # 'ctrl-a b c': 'my-sweet-command' // This is a binding
+    #
+    # This example means the user can type `ctrl-a` then `b` then `c`, and after
+    # all of those keys are typed, it will dispatch the `my-sweet-command`
+    # command.
+    #
+    # The KeymapManager has a couple member variables to deal with multi-stroke
+    # bindings: `@queuedKeystrokes` and `@queuedKeyboardEvents`. They keep track
+    # of the keystrokes the user has typed. When populated, the state variables
+    # look something like:
+    #
+    # @queuedKeystrokes = ['ctrl-a', 'b', 'c']
+    # @queuedKeyboardEvents = [KeyboardEvent, KeyboardEvent, KeyboardEvent]
+    #
+    # Basically, this `handleKeyboardEvent` function will try to exactly match
+    # the user's keystrokes to a binding. If it cant match exactly, it looks for
+    # partial matches. A partial match might be something like:
+    #
+    # // The binding: 'ctrl-a b c': 'my-sweet-command'
+    # @queuedKeystrokes = ['ctrl-a', 'b'] // The user's keystrokes
+    #
+    # When it finds partially matching bindings, it will set the KeymapManager
+    # into a pending state via `enterPendingState`.
```

Comment:
It's worth mentioning that in a "pending state", we're waiting on either another keystroke or a timeout.